### PR TITLE
minor simplifications to DiskFile

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

File: src/disk.hpp [hunk 1]
```diff
@@ -42,45 +41,42 @@ class Disk {
...
+    explicit FileDisk(const fs::path &filename)
     {
         filename_ = filename;
 
         // Opens the file for reading and writing
-        f_ = fopen(filename.c_str(), "w+b");
+        f_ = ::fopen(filename.c_str(), "w+b");
+        if (f_ == nullptr) {
+            throw InvalidValueException(
+                "Could not open " + filename.string() + ": " + ::strerror(errno));
```
Comment (arvidn) [inline src/disk.hpp:56]: always report a failure to open a file (in the future, perhaps we should do something similar to when reading or writing fails, to wait 5 minutes and retry).

File: src/disk.hpp [hunk 2]
```diff
@@ -42,45 +41,42 @@ class Disk {
...
 
     FileDisk(FileDisk &&fd)
     {
         filename_ = fd.filename_;
         f_ = fd.f_;
-        fd.f_ = NULL;
+        fd.f_ = nullptr;
     }
 
-    bool isOpen() { return (f_ != NULL); }
```
Comment (arvidn) [inline src/disk.hpp:65]: is open was only used in one place, to check if the file opened successfully, and throw if it didn't. So I moved that directly into the constructor instead.

File: src/disk.hpp [hunk 3]
```diff
@@ -139,22 +139,22 @@ class FileDisk : public Disk {
         } while (amtwritten != length);
     }
 
-    inline std::string GetFileName() override { return filename_.string(); }
+    std::string GetFileName() override { return filename_.string(); }
 
-    inline uint64_t GetWriteMax() const noexcept { return writeMax; }
+    uint64_t GetWriteMax() const noexcept { return writeMax; }
 
-    inline void Truncate(uint64_t new_size) override
```
Comment (arvidn) [inline src/disk.hpp:146]: all these `inline` are redundant. The function bodies are already defined inline

**Review by hoffmang9 — APPROVED** [review]
Looks good to me

**Pull request merged**